### ISSUE TITLE
Trying to build latest release causes git connections I don't understand

### ISSUE THREAD

**Issue by vanfanel** [diasurgical/devilutionX#4586]
**Important information**
Trying to build on Debian Linux 11

**Describe**
Trying to buid DevilutionX from latest release sources (https://github.com/diasurgical/devilutionX/archive/refs/tags/1.4.0.tar.gz) causes the buildsystem to seemingly hang, doing something unspecified on the console.```

~/src/devilutionX-1.4.0/b4# cmake -DVERSION_NUM=1.4.0 -DCMAKE_BUILD_TYPE=Release -DBINARY_RELEASE=ON -DDEBUG=OFF -DASAN=OFF -DCMAKE_CXX_FLAGS="-march=native -mtune=native -O2" ..
-- The C compiler identification is GNU 10.2.1
-- The CXX compiler identification is GNU 10.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- 📚 zlib: dynamic system library
-- 📚 bzip2: dynamic system library
-- Found BZip2: /usr/lib/aarch64-linux-gnu/libbz2.so (found version "1.0.8") 
-- Looking for BZ2_bzCompressInit
-- Looking for BZ2_bzCompressInit - found
-- 📚 SDL2: dynamic system library
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Found SDL_image
-- 📚 SDL_image: dynamic system library
-- Found SDL2_image: /usr/local/include/SDL2  
-- 📚 SDL_audiolib: static library from source
-- Performing Test HAVE_STD_CLAMP
-- Performing Test HAVE_STD_CLAMP - Success
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- 📚 libsodium: dynamic system library
-- Found sodium: /usr/lib/aarch64-linux-gnu/libsodium.so (found version "1.0.18") 
-- Found fmt 7.1.3
-- 📚 libfmt: dynamic system library
-- Found ZLIB: /usr/lib/aarch64-linux-gnu/libz.so (found version "1.2.11")
```
...and there it stays forever, almost an hour for me already.
Looking at TOP, I see these running:

`git-remote-http`

..which seems to indicate it's downloading something, but what?
Can I somehow simply download sources and build them somehow without having CMAKE to download anything as before?

**To Reproduce**
Get the latest sources and try to build them.

**Expected behavior**
Build from sources.

**Comment by vanfanel** [comment]
Building from git sources (git clone https://github.com/diasurgical/devilutionX.git) has the same effects: CMAKE stays downloading something seemingly forever.

**Comment by tsunamistate** [comment]
@vanfanel it's downloading dependencies, that are not vendored, like Zerotier and libzt, asio, Discord SDK, simpleini etc.
And it looks like some dependency website are blocked for you
Could you please provide full CMake/git command that hangs?

You can also try to disable most of these dependencies by disabling Multiplayer (add `-DNONET=ON` when configuring)

**Comment by vanfanel** [comment]
@tsunamistate Thanks for trying to help me clear this up.

The cmake command that hangs is:
`
cmake -DVERSION_NUM=1.4.0 -DCMAKE_BUILD_TYPE=Release -DBINARY_RELEASE=ON -DDEBUG=OFF -DASAN=OFF -DCMAKE_CXX_FLAGS="-march=native -mtune=native -O2" ..`
(I think that is what you meant?)

Also, what is it doing with ASIO (which is a Windows API with zero meaning on GNU/Linux), Discord SDK (ehmm..why does Diablo need Discord??), etc?
Is ZeroTier needed for local multiplayer?

In other words: Can I have local multiplayer without all that stuff I wousld't touch with a 10mts pole? Is there a CMAKE parameter for that? (ie: network but not whatever those do).

**Comment by tsunamistate** [comment]
You are saying git-remote-http hangs, what are its command line parameters?

* ASIO is asynchronous I/O library needed for multiplayer: https://github.com/diasurgical/asio/
* Discord SDK is needed for Discord Rich Presence, it shows your game status (character, level, time played etc.) in Discord. It is disabled in Linux by default.
* Zerotier is alternative for multiplayer over the Internet, which acts like Hamachi. Its library is free to use, no freemium there

**Comment by vanfanel** [comment]
@tsunamistate Thanks for the further clearing on the issue (I was mistaking ASIO with this one: https://en.wikipedia.org/wiki/Audio_Stream_Input/Output).

I am glad that Zerotier is not mandatory for networking.

About the git-remote-http command line, I don't know what's running that, since the CMAKE output doesn't mention it.
Do you know where is that called? It's impossible to find out since it's not directly called (there isn't a single mention to `git-remote-http` or `git-remote` in the sources, according to `grep`).
If I find where is that called, I may be able to echo it's parameters.

**Comment by AJenbo** [comment]
You can see what is being fetched with `-DFETCHCONTENT_QUIET=FALSE`. You can download them manually and place them where it expects them to avoid it downloading them.

> I hope this engine isn't turning into a bloated monster... it was very fine before :(

No need to panic they are optional modules.

You can download the release source package instead if you have issues with downloading individual dependencies via cmake.
https://github.com/diasurgical/devilutionX/releases/latest/download/devilutionx-src.tar.xz

ZeroTier is a big help for users that are unable to connect directly via TCP. Local multiplayer isn't the only use case.

You can see all the build options via `cmake -LA`, you will probably want `DISABLE_ZERO_TIER` in particular, but you can also disable encryption and some other features if you want to minimize you build.

**Comment by vanfanel** [comment]
@AJenbo Thanks! It seems that `DISABLE_ZERO_TIER` does indeed prevent the hang.

But then I get this:
```

-- 📚 googletest: dynamic system library
CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.18/Modules/FindGTest.cmake:205 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
```

Looked at `cmake -LA` as per your recommendation, and I can't see how to disable this Google Test thing- Is that possible?

**Comment by AJenbo** [comment]
```
$ cmake -LA | grep -i test
BUILD_TESTING:BOOL=ON
```

**Comment by AJenbo** [comment]
Alternativly you can also install the latest dependencies:

https://github.com/diasurgical/devilutionX/blob/master/docs/building.md
```bash
sudo apt-get install cmake g++ libsdl2-dev libsodium-dev libpng-dev libbz2-dev libgtest-dev libgmock-dev libsdl2-image-dev libfmt-dev
```

**Comment by vanfanel** [comment]
@AJenbo @ephphatha 

Thanks! Seems that `BUILD_TESTING=OFF` made cmake succeed at last with no extra dependencies.
Thanks for your patience with me, really.
